crypto/nrf_cc310_platform/src/nrf_cc3xx_platform_mutex_zephyr.c

```diff
@@ -227,6 +228,47 @@ static int32_t mutex_unlock_platform(nrf_cc3xx_platform_mutex_t *mutex) {
     return NRF_CC3XX_PLATFORM_SUCCESS;
 }
 
+#elif CONFIG_CC3XX_ATOMIC_LOCK
+
+atomic_t sym_mutex;
+atomic_t asym_mutex;
+atomic_t rng_mutex;
+atomic_t power_mutex;
+atomic_t heap_mutex;
+
+static void mutex_init_platform(nrf_cc3xx_platform_mutex_t *mutex)
+{
+}
+
+static void mutex_free_platform(nrf_cc3xx_platform_mutex_t *mutex)
+{
+}
+
+static int32_t mutex_lock_platform(nrf_cc3xx_platform_mutex_t *mutex)
+{
+    /* Ensure that the mutex param is valid (not NULL) */
+    if (mutex == NULL) {
+            return NRF_CC3XX_PLATFORM_ERROR_PARAM_NULL;
+    }
+
+    return atomic_cas((atomic_t *)mutex, 0, 1) ?
```

Comment:
This will still need to add the test for uninitialized mutexes (i.e. mutexes in contexts that enable their own mutex because of MBEDTLS_THREADING_C)

_Note that the scheme for the flags should probably be "is_set(SOME_FLAG)" instead of == since we will need to add another flag type to indicate distinction between atomic mutex and regular kernel mutex (see different comment)_